lib/git_worktree.rb

```diff
@@ -225,9 +235,17 @@ def mv_dir(old_dir, new_dir)
     current_index.remove_dir(old_dir)
   end
 
-  def checkout(target_directory)
-    tree = lookup_commit_tree
-    @repo.checkout_tree(tree, :target_directory => target_directory, :strategy => :force)
+  def checkout_at(target_directory)
+    GitWorktree.checkout_at(@path, target_directory)
+  end
+
+  def checkout
+    @repo.checkout(@commit_sha || current_branch, :strategy => :force)
+    @repo.submodules.each do |submodule|
+      submodule.init
```

Comment:
I was expecting `fetch_recurse_submodules => true` somewhere in here.

Reply:
That is only an option for here:

https://github.com/libgit2/rugged/blob/master/ext/rugged/rugged_submodule_collection.c#L270-L271

Which is for "updating the `.git/config`... or maybe `.gitmodule` file, not for actually updating the source (or cloning) submodule.

I tried... and hoped that it was though...

Reply:
Interesting...do we have to manually recurse on the checkout then?

Reply:
Yeah, which is why this method calls a `GitWorktree.checkout_at` below.

I just assume nested submodules just incase, but it works fine when there isn't any nested ones, since `.submodules` will return a empty list in the child clone (or the parent if there are no submodules).

Reply:
ohhhh I see now... Thanks!